src/tests/pools/D3MAavePool.t.sol

```diff
@@ -305,4 +395,15 @@ contract D3MAavePoolTest is D3MPoolBaseTest {
     function test_maxDeposit_returns_max_uint() public {
         assertEq(D3MAavePool(d3mTestPool).maxDeposit(), type(uint256).max);
     }
+
+    function setupV3() internal {
+        aavePool = LendingPoolLike(address(new FakeLendingPoolV3(address(adai))));
+        d3mTestPool = address(new D3MAavePool(D3MAavePool.AaveVersion.V3, "", hub, address(dai), address(aavePool)));
+    }
+
+    function test_aavev3_pool() public {
```

Comment:
I'm not sure about the usage of aavev3 in this unit test, as we only check `adai` is fetched correctly.
Is this intended?

Reply:
Also, if we have a unit test for the Aave v3 pool we should probably also have one for the plan.

Reply:
Hmm maybe this was halfway added by accident - I agree it makes no sense. Will add some more comprehensive unit tests with mocks shortly.

Reply:
Updated. These tests really need to be fixed though. The pool is real unit tests and the plan is using live contracts...